torch/testing/_internal/common_device_type.py

```diff
@@ -432,10 +432,10 @@ def filter_desired_device_types(device_type_test_bases, except_for=None, only_fo
     intersect = set(except_for if except_for else []) & set(only_for if only_for else [])
     assert not intersect, f"device ({intersect}) appeared in both except_for and only_for"
 
-    if except_for is not None:
+    if except_for:
```

Comment:
https://github.com/pytorch/pytorch/blob/6aad12b29b4fc07e96cb6267b6765abcaebf5c32/torch/testing/_internal/common_device_type.py#L499-L511

Here `split_if_not_empty` returns `[]` which trips up `x is not None`.